src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateCopyChanges = [];
```

Comment:
[NAMING] - The name `stateCopyChanges` could be improved to better reflect that it holds the history of state changes. A name like `stateHistory` or `statesHistory` would be more descriptive and in line with the task description.